src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: string = 'triangle',
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        'The longest side of a triangle must be less than' +
+          'the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  constructor(
+    public color: string,
+    public radius: number,
+    public shape: string = 'circle',
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
```

Comment:
[CODE STYLE] - The error message for the Circle constructor should be more descriptive. Instead of 'Radius must be greater than 0', consider specifying why the radius is invalid, for example: 'Circle radius must be greater than 0 to form a valid circle'.